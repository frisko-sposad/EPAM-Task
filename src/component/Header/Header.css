.header__container {
  color: white;
  padding: 1rem;
  background-image: linear-gradient(
      rgba(0, 0, 0, 0.8) 0%,
      rgba(0, 0, 0, 0.8) 100%
    ),
    url(https://assets.nflxext.com/ffe/siteui/vlv3/70731dc8-8977-4ed7-9a0a-421f510aea00/8cbb059e-e3a8-4e36-80fe-33b90a88b39f/RU-en-20200914-popsignuptwoweeks-perspective_alpha_website_large.jpg);
}

.resultSort__container {
  display: flex;
  justify-content: space-between;
  padding: 1rem 3rem;
  font-size: 1rem;
  font-weight: 400;
  text-decoration: none;
}

.resultSort span, .resultCount span {
  padding: 0 1rem;
}

.sortItem {
  text-decoration: none;
  color: #000000;
}

.active {
  color: tomato;
  font-weight: 600;
}