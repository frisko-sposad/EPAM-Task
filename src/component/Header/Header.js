import React from 'react';
import Search from './Search/Search';
import MovieInfo from './MovieInfo/MovieInfo';
import style from './Header.module.css';

const Header = () => {
  return (
    <header className={style.Header}>
      <div><Search /></div>
      {/* <div><MovieInfo /></div> */}
    </header>
  );
};

export default Header;
