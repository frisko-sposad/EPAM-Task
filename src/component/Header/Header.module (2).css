.Header {
  display: flex;
  justify-content: center;
  padding: 1rem;  
  background-image: linear-gradient(rgba(0, 0, 0, 0.8) 0%, rgba(0, 0, 0, 0.8) 100%), url(https://assets.nflxext.com/ffe/siteui/vlv3/70731dc8-8977-4ed7-9a0a-421f510aea00/8cbb059e-e3a8-4e36-80fe-33b90a88b39f/RU-en-20200914-popsignuptwoweeks-perspective_alpha_website_large.jpg);
}

.header__container {
  min-width: 1000px;
}

.btn__container {
  display: flex;
  justify-content: space-between;
  padding: 0.5rem;
}

.sort__container {
  display: flex;
  align-items: baseline;    
  width: 100%;
}

.sort__container span {
  padding: 0 1rem 0 0;
}