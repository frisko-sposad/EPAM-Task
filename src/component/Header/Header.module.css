.header__container {
  color: white;
  padding: 1rem;
  background-image: linear-gradient(
      rgba(0, 0, 0, 0.8) 0%,
      rgba(0, 0, 0, 0.8) 100%
    ),
    url(https://assets.nflxext.com/ffe/siteui/vlv3/70731dc8-8977-4ed7-9a0a-421f510aea00/8cbb059e-e3a8-4e36-80fe-33b90a88b39f/RU-en-20200914-popsignuptwoweeks-perspective_alpha_website_large.jpg);
}

.h1 { 
  margin: 0 auto;
  max-width: 1060px;
  padding: 1rem 3rem;
}

.resultSort__container {
  display: flex;
  justify-content: space-between;
  padding: 1rem 3rem;
  font-size: 1rem;
  font-weight: 400;
}

.resultSort span, .resultCount span {
  padding: 0 1rem;
}

.resultSort span:nth-child(3) {
  color: tomato;
}