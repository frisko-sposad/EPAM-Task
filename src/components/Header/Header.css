.header__container {
  color: white;  
  background-image: linear-gradient(
      rgba(0, 0, 0, 0.8) 0%,
      rgba(0, 0, 0, 0.8) 100%
    ),
    url(https://assets.nflxext.com/ffe/siteui/vlv3/70731dc8-8977-4ed7-9a0a-421f510aea00/8cbb059e-e3a8-4e36-80fe-33b90a88b39f/RU-en-20200914-popsignuptwoweeks-perspective_alpha_website_large.jpg);
}
