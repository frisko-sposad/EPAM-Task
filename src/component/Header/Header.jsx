import React from 'react';
import SearchSection from './SearchSection/SearchSection';
import MovieInfo from './MovieInfo/MovieInfo';
import SiteTitle from './SiteTitle/SiteTitle';
import SortResultsSection from './SortResultsSection/SortResultsSection';
import './Header.css';

const Header = ({ isSearchShown, openSearch }) => {
  return (
    <header>
      <section className="header__container">
        <SiteTitle />
        {isSearchShown ? (
          <SearchSection />
        ) : (
          <MovieInfo openSearch={openSearch} />
        )}
      </section>
      {isSearchShown ? (
        <SortResultsSection
          moviesFound="7 movies found"
          releaseDate="release date"
          sortBy="Sort by"
          rating="rating"
        />
      ) : (
        <SortResultsSection filmsBy="Films by " genre="Drama genre" />
      )}
    </header>
  );
};

export default Header;
