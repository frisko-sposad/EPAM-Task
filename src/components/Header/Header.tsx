import React from 'react';
import { connect } from 'react-redux';
import SearchSection from './SearchSection/SearchSection';
import MovieInfo from './MovieInfo/MovieInfo';
import SiteTitle from './SiteTitle/SiteTitle';
import SortResultsSection from './SortResultsSection/SortResultsSection';
import './Header.css';
import { StateType } from '../App.types';

interface HeaderProps {
  isSearchShown: boolean;
  searchQuery: string;
  total: number;
  searchByOption: string;
  sortBy: string;
}

const Header = ({ isSearchShown, searchQuery, total, searchByOption, sortBy }: HeaderProps) => {
  return (
    <header>
      <section className="header__container">
        <SiteTitle />
        {isSearchShown ? <SearchSection searchByOption={searchByOption} /> : <MovieInfo />}
      </section>
      {isSearchShown ? (
        <SortResultsSection
          moviesFound={`${total} movies found`}
          sortBy={sortBy}
          releaseDate="release date"
          sortByTitle="Sort by"
          rating="rating"
        />
      ) : (
        <SortResultsSection filmsBy="Films by " genre={searchQuery} />
      )}
    </header>
  );
};

const mapStateToProps = ({ isSearchShown, total }: Pick<StateType, 'isSearchShown' | 'total'>) => ({
  isSearchShown,
  total,
});

export default connect(mapStateToProps, null)(Header);
