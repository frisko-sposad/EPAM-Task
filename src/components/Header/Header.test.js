import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Testing the openSearch block in Header.', () => {
  it('If State isSearchShown = true, we check the openSearch snapshot', () => {
    const component = shallow(<Header isSearchShown />);
    expect(component).toMatchSnapshot();
  });
});

describe('Testing the closeSearch block in Header.', () => {
  it('If State isSearchShown = false, we check the closeSearch snapshot', () => {
    const component = shallow(<Header isSearchShown={false} />);
    expect(component).toMatchSnapshot();
  });
});
