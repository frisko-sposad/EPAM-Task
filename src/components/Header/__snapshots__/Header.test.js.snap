// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the closeSearch block in Header. If State isSearchShown = false, we check the closeSearch snapshot 1`] = `
<styled.header>
  <SiteTitle />
  <Switch>
    <Route
      path="/film/:id"
    >
      <Connect(MovieInfo) />
      <Connect(SortResultsSection)
        isSearchShown={true}
      />
    </Route>
    <Route
      path="/"
    >
      <Connect(SearchSection) />
      <Connect(SortResultsSection)
        isSearchShown={false}
      />
    </Route>
  </Switch>
</styled.header>
`;

exports[`Testing the openSearch block in Header. If State isSearchShown = true, we check the openSearch snapshot 1`] = `
<styled.header>
  <SiteTitle />
  <Switch>
    <Route
      path="/film/:id"
    >
      <Connect(MovieInfo) />
      <Connect(SortResultsSection)
        isSearchShown={true}
      />
    </Route>
    <Route
      path="/"
    >
      <Connect(SearchSection) />
      <Connect(SortResultsSection)
        isSearchShown={false}
      />
    </Route>
  </Switch>
</styled.header>
`;
